server/src/main/java/org/elasticsearch/index/mapper/DocumentParserContext.java

```diff
@@ -317,10 +345,15 @@ public final void addDynamicMapper(Mapper mapper) {
      * Consists of a flat set of {@link Mapper}s that will need to be added to their respective parent {@link ObjectMapper}s in order
      * to become part of the resulting dynamic mapping update.
      */
-    public final List<Mapper> getDynamicMappers() {
+    public final Map<String, List<Mapper>> getDynamicMappers() {
```

Comment:
Looking at the access patterns, I wonder if we need two methods:

 - one that returns a flat list of the `List<Mapper>` values (which would be list method as it would still return a `List<Mapper>`
 - One that is `getDynamicMappers(String name)` that returns the `List<Mapper>` that match that name in the hashmap.


This would align nicely with the work here: https://github.com/elastic/elasticsearch/pull/96235